### ISSUE TITLE
deadcode.DeadStores false positive with switch

### ISSUE THREAD

**Issue by ljden** [llvm/llvm-project#189327]
```
    const char *state_str = "?";
    switch (state)
    {
        case UNKNOWN:
            state_str = "Unknown";
            break;

        case AUTHENTICATED:
            state_str = "Authenticated";
            break;

        case ASSOCIATED:
            state_str = "Associated";
            break;

        case AUTHORIZED:
            state_str = "Authorized";
            break;
    }
    printf("%s\n", state_str);
```
Results in the following warning:
```
warning: Value stored to 'state_str' during its initialization is never read [deadcode.DeadStores]
|     const char *state_str = "?";
```
However this is only true if `state` is set to one of the named enum values, which is not guarenteed.

Tested on llvm version 20

**Comment by NeKon69** [comment]
Might also be a first good issue
